> ./compressing_reader_test.go:38:17: undefined: os.ReadFile
> ./compressing_reader_test.go:42:10: undefined: io.NopCloser
> ./compressing_reader_test.go:50:18: undefined: io.ReadAll
> ./compressing_reader_test.go:57:17: undefined: io.ReadAll

Looks like your go version is much older than mine. Technically, should not be a problem to go back to io/ioutil for these.

--- a/compressing_reader.go
+++ b/compressing_reader.go
@@ -204,11 +204,10 @@ func (wr *ovWriter) reset(out []byte) bool {
 		wr.ov = wr.ov[ : 0]
 		wr.ovPos = 0
 		wr.dataPos = ovRem
-	}
-
-	if wr.ovPos > 0 {
+	} else if wr.ovPos > 0 {
 		wr.ov = wr.ov[ : 0]
 		wr.ovPos = 0
+		wr.dataPos = 0
 	}
 
 	wr.data = out

--- a/compressing_reader_test.go
+++ b/compressing_reader_test.go
@@ -3,8 +3,7 @@ package lz4_test
 import (
 	"bytes"
 	"fmt"
-	"io"
-	"os"
+	"io/ioutil"
 	"strings"
 	"testing"
 
@@ -35,11 +34,11 @@ func TestCompressingReader(t *testing.T) {
 				option := option
 				t.Parallel()
 
-				raw, err := os.ReadFile(fname)
+				raw, err := ioutil.ReadFile(fname)
 				if err != nil {
 					t.Fatal(err)
 				}
-				r := io.NopCloser(bytes.NewReader(raw))
+				r := ioutil.NopCloser(bytes.NewReader(raw))
 
 				// Compress.
 				zcomp := lz4.NewCompressingReader(r)
@@ -47,14 +46,14 @@ func TestCompressingReader(t *testing.T) {
 					t.Fatal(err)
 				}
 
-				zout, err := io.ReadAll(zcomp)
+				zout, err := ioutil.ReadAll(zcomp)
 				if err != nil {
 					t.Fatal(err)
 				}
 
 				// Uncompress.
 				zr := lz4.NewReader(bytes.NewReader(zout))
-				out, err := io.ReadAll(zr)
+				out, err := ioutil.ReadAll(zr)
 				if err != nil {
 					t.Fatal(err)
 				}

